### PR TITLE
informant: display I/O stats

### PR DESCRIPTION
**Pull request by ordian**
Not sure what is the best way to display the logs hence the draft:
1. Write to a file.
2. Debug log with a target (currently implemented)
3. Something else.

### CODE REVIEW

File: ethcore/verification/src/queue/mod.rs [hunk 1]
```diff
@@ -234,7 +234,7 @@ impl<K: Kind, C> VerificationQueue<K, C> {
 		let number_of_threads = if scale_verifiers {
 			max_verifiers
 		} else {
-			cmp::min(default_amount, max_verifiers)
+			default_amount
```
Comment (ordian) [inline ethcore/verification/src/queue/mod.rs]: this is an unrelated change, sorry

**Comment by dvdplm** [comment]
>     2\. Debug log with a target (currently implemented)

Let's go with this for the time being and see how useful it is. It would be nice to ponder a more general logging overhaul to add more flexibility but this is not the PR for that.

File: ethcore/src/client/client.rs [hunk 2]
```diff
@@ -1104,7 +1104,19 @@ impl Client {
 	/// Get the report.
 	pub fn report(&self) -> ClientReport {
 		let mut report = self.report.read().clone();
-		report.state_db_mem = self.state_db.read().mem_used();
+		let state_db = self.state_db.read();
+		report.state_db_mem = state_db.mem_used();
+		let io_stats = state_db.journal_db().io_stats();
+		report.io_stats = IoStats {
```
Comment (niklasad1) [inline ethcore/src/client/client.rs]: why can't you just do `report.io_stats = io_stats`?

File: Cargo.lock [hunk 3]
```diff
@@ -1583,7 +1597,7 @@ dependencies = [
  "ethcore-sync",
  "ethereum-types",
  "kvdb",
- "kvdb-rocksdb",
+ "kvdb-rocksdb 0.6.0",
```
Comment (ordian) [inline Cargo.lock]: Upgraded `kvdb-rocksdb` to 0.6 to improve cache stats reporting.
unfortunately `kvdb-rocksdb 0.5.0` is still there because of secret-store

**Comment by ordian** [comment]
> Do you know if/when more fine-grained (caching) would be useful

I think the general consensus is that maintaining multiple levels of caches is a bad idea and we should delegate caching and reuse it from the low-level db and fs cache, like rocksdb.
> why not provide all statistics that kvdb provides?

Not all stats are implemented for kvdb-rocksdb
https://github.com/paritytech/parity-common/blob/master/kvdb-rocksdb/src/stats.rs

File: Cargo.lock (hunk 3 above)
Comment (niklasad1) [reply Cargo.lock]: Ok, there is not possible to bump it in secret-store too?

Comment (ordian) [reply Cargo.lock]: There are some breaking changes https://github.com/paritytech/secret-store#secret-store-20

Comment (ordian) [reply Cargo.lock]: I guess it shouldn't build two versions of rocksdb if secretstore feature is disabled (which is the default)

File: ethcore/types/src/client_types.rs [hunk 4]
```diff
@@ -62,6 +62,27 @@ pub struct ClientReport {
 	pub gas_processed: U256,
 	/// Memory used by state DB
 	pub state_db_mem: usize,
+	/// I/O statistics for state DB.
+	pub io_stats: IoStats,
+}
+
+/// I/O statistics.
+#[derive(Default, Debug, Clone, Eq, PartialEq)]
+pub struct IoStats {
```
Comment (niklasad1) [inline ethcore/types/src/client_types.rs]: This type, seems to be the same as [kvdb::IoStats](https://github.com/paritytech/parity-common/blob/master/kvdb/src/io_stats.rs#L21-#L40) without time? 

Why not use `kvdb::IoStats`? It is because it is not `Default` or what?

Comment (ordian) [reply ethcore/types/src/client_types.rs]: To avoid dependency on `kvdb` and decouple types, such that we can only include fields we're interested in.

File: ethcore/types/src/client_types.rs [hunk 5]
```diff
@@ -62,6 +62,27 @@ pub struct ClientReport {
 	pub gas_processed: U256,
 	/// Memory used by state DB
 	pub state_db_mem: usize,
+	/// I/O statistics for state DB.
```
Comment (dvdplm) [inline ethcore/types/src/client_types.rs]:
```suggestion
	/// I/O statistics for the state DB.
```

Comment (dvdplm) [reply ethcore/types/src/client_types.rs]: …but is it true that it is only for state? I don't think so?

Comment (ordian) [reply ethcore/types/src/client_types.rs]: > …but is it true that it is only for state? I don't think so?

Yes, it is
https://github.com/openethereum/openethereum/pull/11523/files#diff-40c7c65265e511d07158f62712f69573R1101

Comment (dvdplm) [reply ethcore/types/src/client_types.rs]: Hmm, I see. :/ So we miss some IO from other sources then?

File: Cargo.lock (hunk 3 above)
Comment (niklasad1) [reply Cargo.lock]: Yeah, I think that is correct
